### PR TITLE
Update the man help file used in packaging

### PR DESCRIPTION
**Pull request by daxian-dbw**
Update the man help file to reflect the name change.
I only changed the executable name. I think we should continue to use `PowerShell` in the rest of file

After this change, you will get `No manual entry for powershell` when running `man powershell`.
This is how it looks like when running `man pwsh`
image

### CODE REVIEW

**Review by SteveL-MSFT — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -50,9 +50,6 @@ DELETE_ME_TO_DISABLE_CONSOLEHOST_TELEMETRY
 # default location for produced nuget packages
 /nuget-artifacts
 
-# generated man files
-/assets/powershell.1*
-
 # resgen output
```
Comment (adityapatwardhan) [inline .gitignore]: Do we need /assets/pwsh.1* here?

Comment (daxian-dbw) [reply .gitignore]: We don't need it. `New-UnixPackage` command will clean up the generated `pwsh.1.gz` file.

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Comment by iSazonov** [comment]
We should fix:
```
 powershell /h

Usage: powershell[.exe] [[-File] <filePath> [args]]
                        [-Command { - | <script-block> [-args <arg-array>]
                                      | <string> [<CommandParameters>] } ]
                        [-ConfigurationName <string>] [-EncodedCommand <Base64EncodedCommand>]
                        [-ExecutionPolicy <ExecutionPolicy>] [-InputFormat {Text | XML}]
                        [-Interactive] [-NoExit] [-NoLogo] [-NonInteractive] [-NoProfile]
                        [-OutputFormat {Text | XML}] [-Version] [-WindowStyle <style>]

       powershell[.exe] -h | -Help | -? | /?
```

**Comment by daxian-dbw** [comment]
Good catch. I will submit a PR shortly.

**Pull request merged**